src/persistance/metadata-writer.ts

```diff
@@ -1,13 +1,30 @@
-import { Metadata } from "../metadata/MetadataGenerable";
-import { META_DATA_PATH_KEY } from "../constants";
+import {Metadata} from "../metadata/MetadataGenerable";
+import {META_DATA_PATH_KEY} from "../constants";
 import fs from "fs";
 
+const DATA_FILE_EXTENSION = "dat";
+
 export default class MetadataWriter {
   writeMetadata(metaData: Metadata): void {
-    const fileName: string = metaData.id + ".dat";
-    const fileLocation =
-      process.env[META_DATA_PATH_KEY] + metaData.type + "/" + fileName;
+    const metadataRootDir = process.env[META_DATA_PATH_KEY];
 
-    fs.writeFileSync(fileLocation, JSON.stringify(metaData.data), "utf-8");
+    const typeFileName = `${metaData.type}.${DATA_FILE_EXTENSION}`;
```

Comment:
Why did you remove the `if-else` check I had done on type? This `metadata.type` is an optional '?' field which can be undefined. This same writer is used by database and graph models, which doesn't have a type. It would write to `undefined.dat` for those cases.